src/Catalyst/VotingRegistration.tsx

```diff
@@ -38,41 +34,65 @@ export const VotingRegistration = () => {
         detachPreviousScreen: false /* https://github.com/react-navigation/react-navigation/issues/9883 */,
       }}
     >
-      <Stack.Screen name="download-catalyst">
-        {() => (
-          <Boundary loading={{fallbackProps: {style: {flex: 1}}}}>
-            <DownloadCatalyst onNext={() => navigation.navigate('display-pin')} />
-          </Boundary>
-        )}
-      </Stack.Screen>
+      {!complete ? (
+        <Stack.Group>
+          <Stack.Screen name="download-catalyst">
+            {() => (
+              <Boundary loading={{fallbackProps: {style: {flex: 1}}}}>
+                <DownloadCatalyst onNext={navigateTo.displayPin} />
+              </Boundary>
+            )}
+          </Stack.Screen>
 
-      <Stack.Screen name="display-pin">
-        {() => <DisplayPin pin={pin} onNext={() => navigation.navigate('confirm-pin')} />}
-      </Stack.Screen>
+          <Stack.Screen name="display-pin">
+            {() => <DisplayPin onNext={navigateTo.confirmPin} pin={pin} />}
+          </Stack.Screen>
 
-      <Stack.Screen name="confirm-pin">
-        {() => <ConfirmPin pin={pin} onNext={() => navigation.navigate('confirm-tx')} />}
-      </Stack.Screen>
+          <Stack.Screen name="confirm-pin">
+            {/**/}
+            {() => <ConfirmPin onNext={navigateTo.confirmTx} pin={pin} />}
+          </Stack.Screen>
 
-      <Stack.Screen name="confirm-tx">
-        {() => (
-          <Boundary loading={{fallbackProps: {style: {flex: 1}}}}>
-            <ConfirmVotingTx onNext={() => navigation.navigate('qr-code')} />
-          </Boundary>
-        )}
-      </Stack.Screen>
-
-      <Stack.Screen name="qr-code" options={{...defaultStackNavigationOptions, headerLeft: () => null}}>
-        {() => (
-          <Boundary loading={{fallbackProps: {style: {flex: 1}}}}>
-            <QrCode onNext={() => resetToTxHistory()} />
-          </Boundary>
-        )}
-      </Stack.Screen>
+          <Stack.Screen name="confirm-tx">
+            {() => (
+              <Boundary loading={{fallbackProps: {style: {flex: 1}}}}>
+                <ConfirmVotingTx
+                  onNext={() => {
+                    setComplete(true)
+                    navigateTo.qrCode()
+                  }}
+                  pin={pin}
+                  onSuccess={setVotingKeyEncrypted}
```

Comment:
pass votingKeyEncrypted to QRCode via props, so only one hook creates voting reg tx